src/NUnitFramework/framework/Interfaces/TNode.cs

```diff
@@ -115,14 +124,75 @@ public string OuterXml
         /// <returns>A TNode</returns>
         public static TNode FromXml(string xmlText)
         {
-            var doc = new XmlDocument();
-            doc.LoadXml(xmlText);
-            if (doc.FirstChild is null)
+            using var stringReader = new StringReader(xmlText);
+            using var reader = XmlReader.Create(stringReader);
+
+            // go to starting point
+            reader.MoveToContent();
+
+            TNode? root = null;
+            TNode? current = null;
+            var parents = new Stack<TNode>();
+
+            while (!reader.EOF)
             {
-                throw new ArgumentException("Cannot parse text as Xml");
+                if (reader.NodeType == XmlNodeType.Element)
+                {
+                    // keep track of previous which will be promoted to parent stack of reader depth changes
```

Comment:
Should this be "if"?
```suggestion
                    // keep track of previous which will be promoted to parent stack if reader depth changes
```

Reply:
Force-pushed typo fix to keep single commit PR.